dask/array/routines.py

```diff
@@ -598,9 +598,10 @@ def histogram(a, bins=None, range=None, normed=False, weights=None, density=None
     >>> h.compute()
     array([5000, 5000])
     """
-    if bins is None or (range is None and bins is None):
-        raise ValueError('dask.array.histogram requires either bins '
-                         'or bins and range to be defined.')
+    if not np.iterable(bins) and (range is None or bins is None):
+        raise ValueError('dask.array.histogram requires either an iterable '
+                         'specifying the bins or the number of bins and a '
+                         'range to be defined.')
```

Comment:
> 'dask.array.histogram requires either an iterable specifying the bins or the number of bins and a range to be defined.'

What about "dask.array.histogram requires range to be provided if bins is provided as a number. Please define range or specify bins as an iterable of bins"

Reply:
Thanks for the feedback @birdsarah! I like "Please define range or specify bins as an iterable of bins" to guide users in the right direction. My only concern is that this check will also catch the case when `bins` is not specified but `range` is, e.g.

```python
hist, edges = da.histogram(data, range=[1, 10])
```

which works with `np.histogram`. In this case, an error message with "dask.array.histogram requires range to be provided if bins is provided as a number..." could be confusing since `bins` isn't specified at all.

Do you think something like "dask.array.histogram requires either specifying bins as an iterable or specifying both a range and the number of bins. If bins is a number, please define range or specify bins as an iterable of bins" is clear enough for both the `da.histogram(data, bins=40)` and the `da.histogram(data, range=[1, 10])` cases?

Reply:
Yes. I actually think you could drop the second sentence.